apps/hpc-ftsadmin/src/app/components/flow-form.tsx

```diff
@@ -0,0 +1,4094 @@
+  'usageYears',
+  'globalClusters',
+  'locations',
+  'plans',
+  'organizations',
+] as const;
+
+const flowValuesForDisplay = [
+  'amountUSD',
+  'flowDate',
+  'decisionDate',
+  'firstReportedDate',
+  'budgetYear',
+  'origAmount',
+  'origCurrency',
+  'exchangeRate',
+  'activeStatus',
+  'restricted',
+  'newMoney',
+  'description',
+  'notes',
+] as const;
+
+let parentValue = '';
+
+export const FlowForm = (props: Props) => {
+  const {
+    currentVersionData,
+    environment,
+    initialValue,
+    isEdit,
+    prevDetails,
+    versionData,
+    isRestricted,
+    errorCorrection,
+    inputEntries,
+    flowId,
+    versionId,
+    initializeInputEntries,
+    rejectInputEntry,
+    currentVersionID,
+    currentFlowID,
+    currentVersionActiveStatus,
+    isPending,
+    isSuperseded,
+    isCancelled,
+    isCancellation,
+    isNewPending,
+    isUpdatePending,
+    canReactive,
+    pendingFieldsallApplied,
+    allFieldsReviewed,
+    pendingVersionV1,
+  } = props;
+  const { confirm } = dialogs;
+  const env = getEnv();
+
+  const collapseFlowObjects = (data: any) => {
+    data.flowObjects = [];
+
+    collapsePerBehavior(data.dest, 'destination');
+    collapsePerBehavior(data.src, 'source');
+
+    function collapsePerBehavior(behaviorArr: any, ref: any) {
+      Object.keys(behaviorArr).forEach((type) => {
+        behaviorArr[type].forEach((obj: any) => {
+          if (
+            obj !== null &&
+            (!Object.prototype.hasOwnProperty.call(obj, 'cleared') ||
+              !obj.cleared)
+          ) {
+            if (type === 'organization') {
+              obj.objectDetail = obj.implementingPartner ? 'partner' : null;
+            }
+
+            const flowObj = {
+              refDirection: ref,
+              objectType: type,
+              objectID: obj.id,
+              behavior: obj.behavior || null,
+              objectDetail: obj.objectDetail,
+            };
+            data.flowObjects.push(flowObj);
+          }
+        });
+      });
+    }
+
+    return data as any;
+  };
+
+  const collapseCategories = (data: any) => {
+    data.categories = [
+      data.flowType,
+      data.flowStatuses,
+      data.contributionTypes,
+      data.method,
+      data.childMethod,
+      data.keywords,
+      data.inactiveReason,
+      data.beneficiaryGroup,
+      data.pendingStatus,
+      data.earmarking !== null && data.earmarking.id,
+    ].filter((category) => category);
+    data.categories = data.categories
+      .map((value: any) => {
+        if (value && value.id) {
+          return value.id;
+        } else if (value[0]) {
+          return value[0].id;
+        }
+      })
+      .filter(function (value: any) {
+        return value;
+      })
+      .map((value: any) => {
+        return parseInt(value);
+      });
+
+    return data;
+  };
+  const [uploadFileFlag, setUploadFileFlag] = useState<boolean>(false);
+  const [uploadFlag, setUploadFlag] = useState<boolean>(false);
```

Comment:
When declaring these states, if you set an initial value like `false` or `true` the type is inferred and there is no need to explicitly declare it like `<boolean>`. The same applied to any type if you provide an initial value of it.
```suggestion
  const [uploadFileFlag, setUploadFileFlag] = useState(false);
  const [uploadFlag, setUploadFlag] = useState(false);
```